fast64_internal/oot/exporter/base.py

```diff
@@ -0,0 +1,129 @@
+            raise PluginError(f"Exported curves should have only one single segment, found {len(curveData.splines)}")
+
+        return True
+
+    def roundPosition(self, position) -> tuple[int, int, int]:
+        """Returns the rounded position values"""
+
+        return (round(position[0]), round(position[1]), round(position[2]))
+
+    def isCurrentHeaderValid(self, headerSettings: OOTActorHeaderProperty, headerIndex: int):
+        """Checks if the an alternate header can be used"""
+
+        preset = headerSettings.sceneSetupPreset
+
+        if preset == "All Scene Setups" or (preset == "All Non-Cutscene Scene Setups" and headerIndex < 4):
+            return True
+
+        if preset == "Custom":
+            for i, header in enumerate(["childDay"] + altHeaderList):
+                if getattr(headerSettings, f"{header}Header") and i == headerIndex:
+                    return True
+
+            for csHeader in headerSettings.cutsceneHeaders:
+                if csHeader.headerIndex == headerIndex:
+                    return True
+
+        return False
+
+    def getPropValue(self, data, propName: str):
+        """Returns a property's value based on if the value is 'Custom'"""
+
+        value = getattr(data, propName)
+        return value if value != "Custom" else getattr(data, f"{propName}Custom")
+
+    def getConvertedTransformWithOrientation(
+        self, transform: Matrix, dataHolder: Object, obj: Object, orientation: Quaternion | Matrix
+    ):
+        relativeTransform = transform @ dataHolder.matrix_world.inverted() @ obj.matrix_world
+        blenderTranslation, blenderRotation, scale = relativeTransform.decompose()
+        rotation = blenderRotation @ orientation
+        convertedTranslation = ootConvertTranslation(blenderTranslation)
+        convertedRotation = ootConvertRotation(rotation)
+
+        return convertedTranslation, convertedRotation, scale, rotation
+
+    def getConvertedTransform(self, transform: Matrix, dataHolder: Object, obj: Object, handleOrientation: bool):
+        # Hacky solution to handle Z-up to Y-up conversion
+        # We cannot apply rotation to empty, as that modifies scale
+        if handleOrientation:
+            orientation = Quaternion((1, 0, 0), radians(90.0))
+        else:
+            orientation = Matrix.Identity(4)
+        return self.getConvertedTransformWithOrientation(transform, dataHolder, obj, orientation)
+
+    def getAltHeaderListCmd(self, altName: str):
+        """Returns the scene alternate header list command"""
+
+        return indent + f"SCENE_CMD_ALTERNATE_HEADER_LIST({altName}),\n"
+
+    def getEndCmd(self):
+        """Returns the scene end command"""
+
+        return indent + "SCENE_CMD_END(),\n"
+
+
+@dataclass
+class Actor:
+    """Defines an Actor"""
+
+    name: Optional[str] = None
+    id: Optional[str] = None
+    pos: list[int] = field(default_factory=list)
+    rot: Optional[str] = None
+    params: Optional[str] = None
```

Comment:
These shouldn't be optional.

Reply:
same as https://github.com/Fast-64/fast64/pull/256#discussion_r1468456826